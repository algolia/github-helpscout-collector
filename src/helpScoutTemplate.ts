import marked from 'marked';

type TextOptions = {
  content: string;
  link: string;
};

export const formatText = ({ content, link }: TextOptions) => {
  const html = marked(content, {
    sanitize: true,
  });

  return (
    html +
    '<br />\n' +
    '-----------------------------------------------------------\n' +
    '<br />\n' +
    '<h3 style="color: #FF4F81;">\n' +
    'ðŸ‘‰  PLEASE ANSWER ON GITHUB THEN CLOSE THIS TICKET  ðŸ‘ˆ\n' +
    '</h3>\n' +
    `<a href="${link}">${link}</a>\n` +
    '<br/>\n' +
    '-----------------------------------------------------------'
  );
};
