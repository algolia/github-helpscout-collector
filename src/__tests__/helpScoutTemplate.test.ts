import { formatText } from '../helpScoutTemplate';

describe('helpScoutTemplate', () => {
  describe('formatText', () => {
    it('expect to return the content with a footer', () => {
      const content = `
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla metus mi, condimentum eget
convallis quis, tempus a neque. Donec sit amet egestas libero. Praesent tempus magna libero
in semper nibh euismod vel. Pellentesque ultricies eu lorem id ultrices. Pellentesque
ultricies rhoncus turpis, ut rutrum nibh aliquet sit amet.
      `.trim();

      const actual = formatText({
        link: 'https://link.github.com',
        content,
      });

      expect(actual).toMatchSnapshot();
    });

    it('expect to return HTML from MD', () => {
      const content = `
# Hello

This is a test with Markdown.

\`\`\`jsx
const App = () => (
  <View>This is the content of the example.</View>
)
\`\`\`
      `.trim();

      const actual = formatText({
        link: 'https://link.github.com',
        content,
      });

      expect(actual).toMatchSnapshot();
    });

    it('expect to return escaped HTML from MD', () => {
      const content = `
# Hello

This is a test with Markdown.

<script>
  console.log('malicious');
</script>

\`\`\`jsx
const App = () => (
  <View>This is the content of the example.</View>
)
\`\`\`
      `.trim();

      const actual = formatText({
        link: 'https://link.github.com',
        content,
      });

      expect(actual).toMatchSnapshot();
    });
  });
});
