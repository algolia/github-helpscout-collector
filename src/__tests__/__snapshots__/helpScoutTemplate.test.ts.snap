// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpScoutTemplate formatText expect to return HTML from MD 1`] = `
"<h1 id=\\"hello\\">Hello</h1>
<p>This is a test with Markdown.</p>
<pre><code class=\\"language-jsx\\">const App = () =&gt; (
  &lt;View&gt;This is the content of the example.&lt;/View&gt;
)</code></pre>
<br />
-----------------------------------------------------------
<br />
<h3 style=\\"color: #FF4F81;\\">
ðŸ‘‰  PLEASE ANSWER ON GITHUB THEN CLOSE THIS TICKET  ðŸ‘ˆ
</h3>
<a href=\\"https://link.github.com\\">https://link.github.com</a>
<br/>
-----------------------------------------------------------"
`;

exports[`helpScoutTemplate formatText expect to return escaped HTML from MD 1`] = `
"<h1 id=\\"hello\\">Hello</h1>
<p>This is a test with Markdown.</p>
<p>&lt;script&gt;
  console.log('malicious');
&lt;/script&gt;

</p>
<pre><code class=\\"language-jsx\\">const App = () =&gt; (
  &lt;View&gt;This is the content of the example.&lt;/View&gt;
)</code></pre>
<br />
-----------------------------------------------------------
<br />
<h3 style=\\"color: #FF4F81;\\">
ðŸ‘‰  PLEASE ANSWER ON GITHUB THEN CLOSE THIS TICKET  ðŸ‘ˆ
</h3>
<a href=\\"https://link.github.com\\">https://link.github.com</a>
<br/>
-----------------------------------------------------------"
`;

exports[`helpScoutTemplate formatText expect to return the content with a footer 1`] = `
"<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla metus mi, condimentum eget
convallis quis, tempus a neque. Donec sit amet egestas libero. Praesent tempus magna libero
in semper nibh euismod vel. Pellentesque ultricies eu lorem id ultrices. Pellentesque
ultricies rhoncus turpis, ut rutrum nibh aliquet sit amet.</p>
<br />
-----------------------------------------------------------
<br />
<h3 style=\\"color: #FF4F81;\\">
ðŸ‘‰  PLEASE ANSWER ON GITHUB THEN CLOSE THIS TICKET  ðŸ‘ˆ
</h3>
<a href=\\"https://link.github.com\\">https://link.github.com</a>
<br/>
-----------------------------------------------------------"
`;
